src/metabase/lib/breakout.cljc

```diff
@@ -55,6 +43,18 @@
                     (-> (lib.metadata.calculation/metadata query stage-number field-ref)
                         (assoc :lib/source :source/breakouts)))))))
 
+(mu/defn breakout :- ::lib.schema/query
+  "Add a new breakout on an expression, presumably a Field reference. Ignores attempts to add a duplicate breakout."
+  ([query expr]
+   (breakout query -1 expr))
+  ([query        :- ::lib.schema/query
+    stage-number :- :int
+    expr         :- some?]
+   (let [expr (if (fn? expr) (expr query stage-number) expr)]
+     (if (lib.schema.util/distinct-refs? (map lib.ref/ref (cons expr (breakouts query stage-number))))
+       (lib.util/add-summary-clause query stage-number :breakout expr)
+       query))))
+
```

Comment:
I wonder if someone can not have somewhere saved some queries from the past which violate this. No?